### DISCUSSION TITLE
Godot 4.0 gdscript hightlighting doesn't looks great

### DISCUSSION THREAD

**Discussion by Multirious** [helix-editor/helix#6179] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/77918086/222875242-0846b479-2bf2-4100-8d43-36f1e16d038d.png)
This is what I'm currently seeing right now.

I'm expecting it to looks something like this.
Where they are more colorful and more highlighting.
(Also the VSCode version also looks great. I think I got the theme out of the box with godot-tools I guess)
![image](https://user-images.githubusercontent.com/77918086/222875276-ada8052a-3d23-4ed8-a3dd-3e39936cef74.png)

Missing highlighting that I'm expecting:

-  Array static type hint
-  Set get
- Signal parameter
- @ keyword
- Array indexing

I'm currently using theme `catppuccin_mocha` but any other themes looks about the same as the one I'm using.
Godot version is 4.0 stable. The grammar and the highlight query is from the master branch.

If this is need to be manually edit to the theme then please instruct me. Thanks in advance.

**Comment by erasin** [comment]
It still need to wait for https://github.com/PrestonKnopp/tree-sitter-gdscript/issues/9 to finish, and will update the tree-sitter queries file `highlights.scm`.

In addition, `highlights.scm` need to set more detailed settings according to the [theme](https://docs.helix-editor.com/themes.html), for example, the keyword needs to be set at a smaller granularity.

Finally, you can customize `theme-name.toml`

I'll update the queries later.

![WX20230305-113651@2x](https://user-images.githubusercontent.com/716514/222940333-c1cca458-009e-42df-bdf3-3c4bbc1e0ee1.png)

**Comment by erasin** [comment, marked as answer]
highlights for godot4 has merged. https://github.com/helix-editor/helix/pull/6186
the theme can be found in onelight. https://github.com/helix-editor/helix/pull/6192 ,we can define own style based on some theme.

**Reply by Multirious** [reply]
I just update my runtime to latest commit just right now and the highlighting seems to be broken. I'm not sure what's the problem.
![2023-03-10_15-47-57](https://user-images.githubusercontent.com/77918086/224269421-0b953ba1-8d56-437a-b73b-be56bccaf9f9.gif)

**Reply by erasin** [reply]
Did you update the grammar? 

hx -g fetch && hx -g build 
  -g, --grammar {fetch|build}    Fetches or builds tree-sitter grammars listed in languages.toml


``` 
hx -g fetch && hx -g build 
```

---

If you are using hx compiled by master branch, then it will automatically fetch and build grammar when cargo build the hx.

If you are using a downloaded release version, after updating the files in `runtime/quires/gdscript/*.scm` , you need to set the git rev in the custom configuration `.config/languages.toml`.

```
[[grammar]]
name = "gdscript"
source = { git = "https://github.com/PrestonKnopp/tree-sitter-gdscript", rev = "a4b57cc3bcbfc24550e858159647e9238e7ad1ac" }
```

and update grammar

``` 
hx -g fetch && hx -g build 
```

**Reply by Multirious** [reply]
yes, I did just that when I cloned the runtime folder. Though, I haven't set the git rev. Let me try again.

**Reply by Multirious** [reply]
Working now. Thanks for your work!
